daemon/cmd/deletion_queue.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/lock/lockfile"
+)
+
+// processQueuedDeletes is the agent-side of the identity deletion queue.
+// The CNI plugin queues deletions when the agent is down, because
+// containerd / crio expect CNI DEL to always succeed. It does so
+// by writing a file to /run/cilium/deleteQueue with the endpoint ID
+// to delete.
+//
+// On startup, we will grab a lockfile in this directory, then process
+// all deletions. Then, we start up the agent server, then drop the lock.
+// Any CNI processes waiting in that period of time will, after getting
+// the lock.
+//
+// Returns a done function that drops the lock. It should be called
+// after the server is running.
+func (d *Daemon) processQueuedDeletes() func() {
+	if err := os.MkdirAll(defaults.DeleteQueueDir, 0755); err != nil {
+		log.WithError(err).WithField("path", defaults.DeleteQueueDir).Error("Failed to ensure deletion queue directory exists")
+		return func() {}
+	}
+
+	log.Infof("Processing queued endpoint deletions")
+
+	var lf *lockfile.Lockfile
+	locked := false
+
+	unlock := func() {
+		if lf != nil && locked {
+			lf.Unlock()
+		}
+		if lf != nil {
+			lf.Close()
+		}
+	}
+
+	lf, err := lockfile.NewLockfile(defaults.DeleteQueueLockfile)
+	if err != nil {
+		log.WithError(err).WithField("filename", defaults.DeleteQueueLockfile).Error("Failed to create ")
```

Comment:
Nit: we already have a logfield for file paths, we should prefer this one for the sake of consistency especially in structured logging. Please also apply for each occurrence.

```suggestion
		log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueLockfile).Error("Failed to create ")
```

Reply:
Also, upon a second read, I notice that the error logs are quite terse. I think we'd benefit from providing more context when there are warnings / errors, because we don't want the user to then think, "OK well I've got this error, but what should I do / what is Cilium going to do about it?".

This one feels more like a warning because Cilium is still proceeding anyway. 

```suggestion
		log.WithError(err).WithField("filename", defaults.DeleteQueueLockfile).Warn("Failed to create CNI delete directory queue lockfile. Proceeding with deletion anyway")
```